vllm-proxy/tests/signature_verification_example.py

```diff
@@ -0,0 +1,304 @@
+#!/usr/bin/env python3
```

Comment:
Nice! This is super helpful. Can you make it organized in this way since it's clearer? It has a few advantages:

- Each example is complete
- They just need to focus on one example first. Don't need to read the unrelated code.
- `verify_signature_for_chat()` can be a pure verification function

```
def exmaple_nonstream_direct():
   request = nonstream_request()
   response = request.get(....)
   signature = request.get(...)
   verify_signature_for_chat(request, response, signature)

def exmaple_nonstream_routed():
   request = nonstream_request()
   response = request.get(....)
   signature = request.get(...)
   verify_signature_for_chat(request, response, signature)


def exmaple_stream_direct():
   pass

def exmaple_nonstream_routed():
   pass
```